File: listas_pc/lista-1/G.cpp
#include<bits/stdc++.h>
#define desync ios::sync_with_stdio(false); cin.tie(NULL); cout.tie(NULL);
using namespace std;

int solve(string x){
    map<string, int> dic;
    string aux;
    for(int i = 0; i < sizeof(x); i++){
        cout << x[i] << "\n";
        aux = x[i];
        if (dic.find(aux) == dic.end()){
            dic[aux] = 1;
        }
        else{
            return false;
        }
    }
    return true;

}

int main(){
    int n;
    cin >> n;
    
    n++;
    string m = to_string(n);
    
    cout << solve(m) << "\n";


    /*while (solve(m) != true){
        n = stoi(m);
        n++;
        m = to_string(n);
        cout << "n " << n << "m " << m;
    }
    cout << n << "\n";*/
    return 0;
}

File: ups_semanauni/G.py
def solve():
    a = input()
    b = input()

    counter = 0
    dic = {}
    for i in range(len(a)):
        if a[i] == b[i] and a[i] not in dic:
            dic[a[i]] = b[i]
        elif a[i] != b[i] and a[i] not in dic:
            dic[a[i]] = b[i]
            counter+= 1
        elif a[i] == b[i] and a[i] in dic:
            if b[i] != dic[a[i]]:
                return -1
        elif a[i] != b[i] and a[i] in dic:
            return -1

    return counter

n = int(input())

print(solve())